src/transforms/remap.rs

```diff
@@ -40,14 +40,18 @@ const DROPPED: &str = "dropped";
 /// Configuration for the `remap` transform.
 #[configurable_component(transform("remap"))]
 #[derive(Clone, Debug, Derivative)]
-#[serde(deny_unknown_fields, default)]
```

Comment:
Telling `serde` to use a default value for deserializing the configuration of a typed component makes no sense, because these types are never a fixed field... only part of a field using their containing enums i.e. `Transforms`.

It does, however, lead to the `vector-config-macros` code thinking that there's a default value that should be added to the configuration schema, which again... does not make sense at _this_ particular level.